build-system/compile/compile.js

```diff
@@ -36,7 +37,27 @@ const {VERSION: internalRuntimeVersion} = require('../internal-version');
 const isProdBuild = !!argv.type;
 const queue = [];
 let inProgress = 0;
-const MAX_PARALLEL_CLOSURE_INVOCATIONS = argv.single_pass ? 1 : 4;
+// NOTE(erwinm): Theres a weird race condition where when we use `pseudo_names`
```

Comment:
I tried solving this race for single pass a few different ways in #22040, but had to abandon that PR. Can you file an issue and add a `TODO`? I'll take another swipe at fixing it.

Reply:
done.